### PR TITLE
Cloudscale Regions and Zones Support

### PR DESCRIPTION
**Pull request by href**
##### SUMMARY
Adds support for [cloudscale.ch](https://cloudscale.ch)'s regions and zones.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- `cloudscale_floating_ip`
- `cloudscale_server`
- `cloudscale_server_group`
- `cloudscale_volume`

##### ADDITIONAL INFORMATION
This change brings our Ansible modules closer to feature parity with our official REST api. Additionally it includes some spelling fixes.

This is my first change to Ansible, please feel free to request additional adjustments to my PR.

### CODE REVIEW

**Comment by href** [comment]
@resmo I was told to ping you on this by my supervisor 😉

File: lib/ansible/modules/cloud/cloudscale/cloudscale_server.py [hunk 1]
```diff
@@ -211,13 +219,19 @@
 flavor:
   description: The flavor that has been used for this server
   returned: success when not state == absent
-  type: str
-  sample: flex-8
+  type: dict
+  sample: { 'slug': 'flex-4', 'name': 'Flex-4', 'vcpu_count': 2, 'memory_gb': 4}
```
Comment (resmo) [inline lib/ansible/modules/cloud/cloudscale/cloudscale_server.py]: I would suggest to move this "fix" to a separate PR, and IMHO the same fix would be required for "image" as well.

Comment (href) [reply lib/ansible/modules/cloud/cloudscale/cloudscale_server.py]: Sure, I reverted 9e7c798 and I'll follow up with another PR that fixes these documentation issues. Thanks!

File: lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py [hunk 2]
```diff
@@ -137,6 +154,11 @@
   returned: success
   type: str
   sample: 185.98.122.176
+region:
+  description: The region of the floating IP.
+  returned: success when state == present
+  type: dict
+  sample: {'slug': 'lpg'}
```
Comment (resmo) [inline lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py:161]: please add the `version_added: '2.10'` here as well

File: lib/ansible/modules/cloud/cloudscale/cloudscale_server_group.py [hunk 3]
```diff
@@ -89,6 +102,11 @@
   returned: if available
   type: str
   sample: anti-affinity
+zone:
+  description: The zone of the server group
+  returned: success
+  type: dict
+  sample: { 'slug': 'rma1' }
```
Comment (resmo) [inline lib/ansible/modules/cloud/cloudscale/cloudscale_server_group.py:109]: please add the `version_added: '2.10'`here as well

File: lib/ansible/modules/cloud/cloudscale/cloudscale_volume.py [hunk 4]
```diff
@@ -138,6 +147,11 @@
   returned: state == present
   type: str
   sample: bulk
+zone:
+  description: The zone of the volume.
+  returned: state == present
+  type: dict
+  sample: {'slug': 'lpg1'}
```
Comment (resmo) [inline lib/ansible/modules/cloud/cloudscale/cloudscale_volume.py:154]: please add the `version_added: '2.10'`here as well

File: lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py (hunk 2 above)
Comment (href) [reply lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py:161]: Sure, sorry for not providing that up front.

**Comment by href** [comment]
Note to self: Do not use GitHub's rebase feature. Somehow it ended up producing garbage-y results in this case.

@resmo I fixed the rebase and it should now be ready to merge.

File: test/integration/targets/cloudscale_server/tasks/tests.yml [hunk 5]
```diff
@@ -42,6 +45,7 @@
       - server is changed
       - server.state == 'running'
       - server.server_groups.0.name == '{{ cloudscale_resource_prefix }}-group-1'
+      - server.zone.slug == '{{ cloudscale_test_zone }}'
```
Comment (resmo) [inline test/integration/targets/cloudscale_server/tasks/tests.yml:48]: not a blockerl, just fyi: the templating is superfluous --> `server.zone.slug == cloudscale_test_zone`

**Review by resmo — APPROVED** [review]
shipit

File: test/integration/targets/cloudscale_server/tasks/tests.yml (hunk 5 above)
Comment (href) [reply test/integration/targets/cloudscale_server/tasks/tests.yml:48]: Ah, that makes sense, thanks. I'll keep that in mind for future changes.

**Comment by href** [comment]
Nice, thanks 🎉

**Pull request merged**